compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1548,9 +1551,66 @@ impl<'a> Parser<'a> {
             Ok(self.mk_expr_err(lo))
         } else {
             let msg = "expected `while`, `for`, `loop` or `{` after a label";
-            self.struct_span_err(self.token.span, msg).span_label(self.token.span, msg).emit();
+
+            let mut err = self.struct_span_err(self.token.span, msg);
+            err.span_label(self.token.span, msg);
+
             // Continue as an expression in an effort to recover on `'label: non_block_expr`.
-            self.parse_expr()
+            let expr = self.parse_expr().map(|expr| {
```

Comment:
Can you actually use `?` here, move the body of the closure to below, and remove the `?` on line 1614?

Reply:
All other if branches also return `Result`s, so I don't think it makes sense to do that

Reply:
Oh, oops, I thought there was only two if branches. I should've clicked the button to reveal the lines above :sweat: